packages/@o3r/eslint-plugin/src/rules/o3r-widget-tags/o3r-widget-tags.ts

```diff
@@ -0,0 +1,168 @@
+import { createRule } from '../utils';
+
+const o3rWidgetParameterPattern = '^[a-zA-Z0-9-_:.]+$';
+
+export interface O3rWidgetTagsRuleOption {
+  [widgetName: string]: {
+    [paramName: string]: 'string' | 'number' | 'boolean' | 'string[]' | 'number[]' | 'boolean[]';
+  };
+}
+
+type O3rWidgetRuleErrorId = 'notSupportedType' | 'notSupportedParamForType' | 'invalidParamValueType' | 'noParamWithoutWidget' | 'duplicatedWidget' | 'duplicatedParam';
+
+export default createRule<O3rWidgetTagsRuleOption[], O3rWidgetRuleErrorId>({
+  name: 'o3r-widget-tags',
+  meta: {
+    hasSuggestions: false,
+    fixable: undefined,
+    type: 'problem',
+    docs: {
+      description: 'Ensures that @o3rWidget and @o3rWidgetParam are used with correct value',
+      recommended: 'error'
+    },
+    schema: [
+      {
+        type: 'object',
+        additionalProperties: {
+          type: 'object',
+          additionalProperties: false,
+          patternProperties: {
+            [o3rWidgetParameterPattern]: {
+              type: 'string',
+              enum: [
+                'string',
+                'number',
+                'boolean',
+                'string[]',
+                'number[]',
+                'boolean[]'
+              ]
+            }
+          }
+        }
+      }
+    ],
+    messages: {
+      notSupportedType: '{{ o3rWidgetType }} is not supported. Supported o3rWidget types: {{ supportedO3rWidgets }}.',
+      notSupportedParamForType: '{{ o3rWidgetParam }} is not supported for {{ o3rWidgetType }}. Supported o3rWidgetParam for {{ o3rWidgetType }}: {{ supportedO3rWidgetParam }}.',
+      invalidParamValueType: '{{ o3rWidgetParam }} supports only {{ o3rWidgetParamType }}.',
+      noParamWithoutWidget: '@o3rWidgetParam could not be used without @o3rWidget',
+      duplicatedParam: '@o3rWidgetParam {{ o3rWidgetParam }} must be defined only once.',
+      duplicatedWidget: '@o3rWidget must be defined only once.'
+    }
+  },
+  defaultOptions: [],
+  create: (context) => {
+    const options: O3rWidgetTagsRuleOption = context.options.reduce((acc, option) => ({ ...acc, ...option }), {});
+    const supportedO3rWidgets = new Set(Object.keys(options));
+    return {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      TSPropertySignature: (node) => {
```

Comment:
This should be restricted to properties in an interface extending `Configuration` (and maybe raise an error if it is not the case)